generative_models/DCGAN_pytorch/main.py

```diff
@@ -0,0 +1,109 @@
+import torch
+import torchvision
+import torch.nn as nn                       # All neural network modules, nn.Linear, nn.Conv2d, BatchNorm, Loss functions
+import torch.optim as optim                 # For all Optimization algorithms, SGD, Adam, etc.
+import torchvision.datasets as datasets     # Has standard datasets we can import in a nice way
+import torchvision.transforms as transforms # Transformations we can perform on our dataset
+from torch.utils.data import DataLoader     # Gives easier dataset managment and creates mini batches
+
+import argparse
+parser = argparse.ArgumentParser()
+parser.add_argument('--image_size', type=int, default=64)
+parser.add_argument('--num_epochs', type=int, default=10)
+parser.add_argument('--batch_size', type=int, default=128)
+parser.add_argument('--features_d', type=int, default=128)
+parser.add_argument('--features_g', type=int, default=128)
+parser.add_argument('--channels_noise', type=int, default=100)
+parser.add_argument('--lr', type=float, default=0.0002)
+parser.add_argument('--beta1', type=float, default=0.5)
+args = parser.parse_args()
+
+lr = args.lr
+batch_size = args.batch_size
+image_size = args.image_size
+channels_noise = args.channels_noise
+num_epochs = args.num_epochs
+features_d = args.features_d
+features_g = args.features_d
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+my_transforms = transforms.Compose([
+    transforms.Resize(image_size),
+    transforms.ToTensor(),
+    transforms.Normalize((0.5,),(0.5,)),
+    ])
+
+dataset = datasets.MNIST(root='dataset/', train=True, transform=my_transforms, download=True)
+dataloader = DataLoader(dataset, batch_size=batch_size, shuffle=True, num_workers=2)
+
+from models import Generator , Discriminator , weights_init
+netD = Discriminator(channels_img, features_d).to(device)
+netG = Generator(channels_noise, channels_img, features_g).to(device)
+netG=netG.apply(weights_init)
+netD=netD.apply(weights_init)
+
+optimizerD = optim.Adam(netD.parameters(), lr=lr, betas=(0.5, 0.999) )
+optimizerG = optim.Adam(netG.parameters(), lr=lr, betas=(0.5, 0.999) )
```

Comment:
use beta1 args

Reply:
didnt understand , what about beta1 args?

Reply:
ok got it

Reply:
dekhlo sare changes bana diye

Reply:
also i had picked wGAN but i am submitting dcgan because i was new to gans so couldnt get the hang of it